homeassistant/components/template/coordinator.py

```diff
@@ -104,6 +118,10 @@ async def _attach_triggers(self, start_event: Event | None = None) -> None:
     async def _handle_triggered_with_script(
         self, run_variables: TemplateVarsType, context: Context | None = None
     ) -> None:
+        # Render run variables after the trigger, before checking conditions.
+        if self._run_variables:
+            run_variables = self._run_variables.async_render(self.hass, run_variables)
```

Comment:
This too should be covered by a test

Reply:
It should be covered by the blueprint test, I can make an exclusive test for it

Reply:
Yep, it's missing.  The test was covering variables being resolved between trigger and action, this like of code is actually executed when the scripts run.  Brain fart on my part.